### ISSUE TITLE
Support for shearing of lines and inline elements

### ISSUE THREAD

**Issue by palemieux** [w3c/csswg-drafts#2983]
Japanese subtitles commonly use shearing of text, as illustrated below.

Not sheared:

![plain](https://user-images.githubusercontent.com/4871350/43588395-b2363338-966c-11e8-9389-31d84b62b9cd.PNG)

Sheared:

![shear](https://user-images.githubusercontent.com/4871350/43588411-b93752ca-966c-11e8-873c-4bde37983fd0.PNG)

Couple of observations:

* while shearing might not be appropriate for books, it is commonly used for subtitles

* shearing is not applied to individual characters, but instead to complete blocks -- each of the digits in 1998 is not sheared individually

* the shear angle is author-specified

* it should be possible to shear individual lines as well as groups of lines

CSS Transforms Module Level 1 `skewX()` and `skewY()` can be applied to blocks but neither to lines nor inline elements

**Comment by astearns** [comment]
I believe we've been intending to support this with `font-style: -20%` (or whatever value gives you the synthesized oblique you require)

https://drafts.csswg.org/css-fonts-4/#font-style-prop

**Comment by palemieux** [comment]
@astearns I read `font-style: -20%` as applying to each character individually. Per the illustrations above, the shear is applied to ruby base and text as a block.

**Comment by astearns** [comment]
We might be able to handle that by defining the desired outcome for upright characters, text-combine-upright runs and ruby text when synthesized oblique is applied.

**Comment by kojiishi** [comment]
Can I ask where you got the screenshot? I wonder how much common to shear ruby text together. AFAIK, it is not common in publishing world. Maybe different in TTML usages?

**Comment by palemieux** [comment]
> AFAIK, it is not common in publishing world. Maybe different in TTML usages?

Yes, as I understand it, shearing is not common in printing/publishing, but common in subtitles.

Below is a selection of pointers:

- thread at https://github.com/w3c/ttml2/issues/784
- [SMPTE ST 428-7](https://doi.org/10.5594/SMPTE.ST428-7.2014), which is the subtitling format used worldwide for cinema
- [SST G1 manual](http://canvass.co.jp/global/common/pdf/sstg1-operationmanual.pdf)

>Can I ask where you got the screenshot? 

The screen shot was generated by [imscJS](https://github.com/sandflow/imscJS), which an implementation of IMSC 1.1. See the [sample app](http://sandflow.com/imsc1_1/index.html) and the following IMSC 1.1 sample.

```
<?xml version="1.0" encoding="UTF-8"?>
<tt xmlns:tt="http://www.w3.org/ns/ttml"
    xmlns:ttp="http://www.w3.org/ns/ttml#parameter"
    xmlns="http://www.w3.org/ns/ttml"
    xmlns:tts="http://www.w3.org/ns/ttml#styling"
    ttp:contentProfiles="http://www.w3.org/ns/ttml/profile/imsc1.1/text">
  <head>

    <styling>

      <style xml:id="s1" tts:showBackground="whenActive" tts:color="white" tts:writingMode="tbrl" tts:fontSize="150%"/>

    </styling>

    <layout>

      <region style="s1" xml:id="vertical" tts:origin="55% 5%" tts:extent="40% 90%"/>

    </layout>

  </head>

  <body>

    <div begin="0s" end="1s" tts:shear="16.67%" region="vertical">

      <p>
        <span tts:ruby="container" tts:rubyAlign="spaceAround" tts:rubyPosition="before">
          <span tts:ruby="baseContainer"><span tts:ruby="base">山</span></span><span tts:ruby="textContainer"><span tts:ruby="text">やま</span></span></span>の
      </p>
      <p>
        <span tts:ruby="container" tts:rubyAlign="spaceAround" tts:rubyPosition="after">
         <span tts:ruby="baseContainer"><span tts:ruby="base">最多<span tts:textCombine="all">1998</span>年</span></span>
          <span tts:ruby="textContainer"><span tts:ruby="text">ルビサンプル</span></span>
        </span>
      </p>

    </div>

  </body>

</tt>
```

**Comment by kojiishi** [comment]
> Yes, as I understand it, shearing is not common in printing/publishing, but common in subtitles.

Sorry, it looks like I mislead you.

Shearing is common in printing/publishing. We've been discussing how to do this in css-fonts for years.

When it has ruby, it is common to shear ruby base and ruby text separately in printing/publishing, and that's the only portion where we see TTML has different requirement from printing/publishing.

So the question is, is shifting ruby text position an important factor to need a separate mode for TTML? Do you have any pointers that indicates ruby positioning of sheared text should be different from printing/publishing?

**Comment by kojiishi** [comment]
For `text-combine-upright`, it's a good point. The requirement is same, we should fix it in css-fonts.

**Comment by palemieux** [comment]
> So the question is, is shifting ruby text position an important factor to need a separate mode for TTML? Do you have any pointers that indicates ruby
> positioning of sheared text should be different from printing/publishing?

See w3c/ttml2#784 for some pointers.

@akiseino who has direct experience with cinema subtitling should be able to provide additional evidence.

**Comment by kojiishi** [comment]
I see @akiseino says on 23 May:
> The ruby by individual, is moved to left side. But it's center by block.

It indicates ruby should be centered, not to be moved by shearing. IIUC this is the same as printing/publishing.

**Comment by palemieux** [comment]
> It indicates ruby should be centered, not to be moved by shearing. IIUC this is the same as printing/publishing.

Can you provide an illustration to avoid any misunderstanding?

**Comment by svgeesus** [comment]
Related to [oblique angle for synthesis in vertical text ](https://github.com/w3c/csswg-drafts/issues/2869)

**Comment by litherum** [comment]
See also: https://github.com/w3c/csswg-drafts/issues/4044

**Comment by cconcolato** [comment]
What should be done to progress the support for line-shearing in CSS?

**Comment by BHolterMedia** [comment]
Hi all just adding my $.02 since I just discovered this issue and how it impacts shearing of Japanese subtitles in CSS scenarios. This impacts all Japanese subtitles we author and while many of the use cases do not depend on CSS, many do and require workarounds to accomplish on screen what has been authored in the ttml itself. For my use case specifically the region extents cropping text when shearing is applied, requiring modification to the style settings that we otherwise wouldn't want to include. See below for an example of shearing being cropped as displayed via imscJS CSS.

![image](https://github.com/user-attachments/assets/64cb5c8e-53f2-4bcb-abac-1f2dbf22f997)

For more media context, more and more content owners and media platforms are implementing IMSC as their core timed text format for media inventories and mezzanine time text distribution assets, but many platforms still perform format conversions (converting from IMSC to VTT, etc.) to display subtitles due to limitations like this (and of course for other reasons internal to their own media processing pipelines).
